lib/tom_queue/deferred_work_manager.rb

```diff
@@ -87,7 +87,7 @@ def start
       # (which have been scheduled by the AMQP consumer). If a message is returned
       # then we re-publish the messages to our internal QueueManager and ack the deferred
       # message
-      until @shutdown
+      while true
```

Comment:
I'd prefer `loop do`, but that changes the scope of variables, etc. I don't _think_ that'll be an issue here, but I'm not bothered if we have to leave it as `while true`.